x/evm/keeper/state_transition.go

```diff
@@ -70,17 +70,24 @@ func (k *Keeper) NewEVM(
 	if tracer == nil {
 		tracer = k.Tracer(msg, cfg.ChainConfig)
 	}
-	vmConfig := k.VMConfig(msg, cfg.Params, tracer)
+	vmConfig := k.VMConfig(cfg.Params, tracer)
 	return vm.NewEVM(blockCtx, txCtx, k, cfg.ChainConfig, vmConfig)
 }
 
 // VMConfig creates an EVM configuration from the debug setting and the extra EIPs enabled on the
 // module parameters. The config generated uses the default JumpTable from the EVM.
-func (k Keeper) VMConfig(msg core.Message, params types.Params, tracer vm.Tracer) vm.Config {
+func (k Keeper) VMConfig(params types.Params, tracer vm.Tracer) vm.Config {
 	fmParams := k.feeMarketKeeper.GetParams(k.Ctx())
 
+	var debug bool
+	if _, ok := tracer.(types.NoOpTracer); ok {
+		debug = false
+	} else {
+		debug = true
+	}
+
 	return vm.Config{
-		Debug:       k.debug,
```

Comment:
Let's remove this from the keeper fields too

Reply:
I thought about removing it.. but this is still being use for this https://github.com/tharsis/ethermint/blob/main/x/evm/types/tracer.go#L27  in here https://github.com/tharsis/ethermint/blob/main/x/evm/keeper/keeper.go#L386.

Would you like me to change that?

Reply:
yes, let's remove it